x-pack/elastic-agent/pkg/core/plugin/process/status.go

```
@@ -35,21 +42,74 @@ func (a *Application) OnStatusChange(s *server.ApplicationState, status proto.St
 			return
 		}
 
-		// kill the process
-		if a.state.ProcessInfo != nil {
-			_ = a.state.ProcessInfo.Process.Kill()
-			a.state.ProcessInfo = nil
-		}
-		ctx := a.startContext
-		tag := a.tag
-
 		// it was marshalled to pass into the state, so unmarshall will always succeed
 		var cfg map[string]interface{}
 		_ = yaml.Unmarshal([]byte(s.Config()), &cfg)
```

Unrelated to this PR but I don't think we should swallow the errors here.